### ISSUE TITLE
[Bug]: Getting empty results when querying with id that was deleted but inserted again

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#16417]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20220407-e2eed6fd
- Deployment mode(standalone or cluster): standalone
- SDK version(e.g. pymilvus v2.0.0rc2): 2.0.1
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. The test case in `test_delete.py`
```
 @pytest.mark.tags(CaseLabel.L2)
    @pytest.mark.parametrize("to_query", [True, False])
    @pytest.mark.parametrize("to_flush", [True, False])
    def test_delete_insert_same_id_growing(self, to_query, to_flush):
        """
        target: test insert same id entity after delete from growing data
        method: 1.create and load
                2.insert entities and no flush
                3.delete id 0 entity
                4.insert new entity with same id
                5.query with the id
        expected: Verify that the query gets the newly inserted entity
        """
        # init collection and insert data without flush
        collection_w = self.init_collection_wrap(name=cf.gen_unique_str(prefix))
        collection_w.load()
        df = cf.gen_default_dataframe_data(tmp_nb)
        collection_w.insert(df)

        # delete
        del_res, _ = collection_w.delete(tmp_expr)
        if to_query:
            collection_w.query(tmp_expr, check_task=CheckTasks.check_query_empty)

        # insert entity with primary key 0
        df_new = cf.gen_default_dataframe_data(1)
        collection_w.insert(df_new)
        if to_flush:
            log.debug(collection_w.num_entities)

        # query entity one
        res = df_new.iloc[[0], [0, 2]].to_dict('records')
        collection_w.query(tmp_expr, output_fields=[ct.default_float_vec_field_name],
                           check_task=CheckTasks.check_query_results, check_items={'exp_res': res, 'with_vec': True})

```
2. The case running results. If you run the test case you will find that the test case will pass if and only if 
 `to_query=True, to_flush=False`. I am desperate to know why.
```
[2022-04-07 17:58:39,002 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None], kwargs: {'timeout': 20} (api_request.py:55)
[2022-04-07 17:58:39,561 - DEBUG - ci_test]: (api_response) : None  (api_request.py:27)
[2022-04-07 17:58:39,595 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [    int64  float                                       float_vector
0       0    0.0  [0.0027491626455215207, 0.06323141541753616, 0...
1       1    1.0  [0.07674878577118355, 0.10358780170919225, 0.0...
2       2    2.0  [0.058777331989754905, 0.10100433064952458, 0....
3       3    3.0  [0.153574......, kwargs: {'timeout': 20} (api_request.py:55)
[2022-04-07 17:58:39,951 - DEBUG - ci_test]: (api_response) : (insert count: 100, delete count: 0, upsert count: 0, timestamp: 432360789086044161)  (api_request.py:27)
[2022-04-07 17:58:39,952 - DEBUG - ci_test]: (api_request)  : [Collection.delete] args: ['int64 in [0]', None, 20], kwargs: {} (api_request.py:55)
[2022-04-07 17:58:39,977 - DEBUG - ci_test]: (api_response) : (insert count: 0, delete count: 1, upsert count: 0, timestamp: 432360789099151362)  (api_request.py:27)
[2022-04-07 17:58:39,977 - DEBUG - ci_test]: (api_request)  : [Collection.query] args: ['int64 in [0]', None, None, 20], kwargs: {} (api_request.py:55)
[2022-04-07 17:58:40,380 - DEBUG - ci_test]: (api_response) : []  (api_request.py:27)
[2022-04-07 17:58:40,389 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [   int64  float                                       float_vector
0      0    0.0  [0.013268341152998242, 0.06522509266242461, 0...., None], kwargs: {'timeout': 20} (api_request.py:55)
[2022-04-07 17:58:40,429 - DEBUG - ci_test]: (api_response) : (insert count: 1, delete count: 0, upsert count: 0, timestamp: 432360789217378308)  (api_request.py:27)
[2022-04-07 17:58:43,083 - DEBUG - ci_test]: 101 (test_delete.py:769)
[2022-04-07 17:58:43,086 - DEBUG - ci_test]: (api_request)  : [Collection.query] args: ['int64 in [0]', ['float_vector'], None, 20], kwargs: {} (api_request.py:55)
[2022-04-07 17:58:43,408 - DEBUG - ci_test]: (api_response) : []  (api_request.py:27)
FAILED
```

**Comment by yanliang567** [comment]
@soothing-rain could u please take a look at this issue?

/assign @soothing-rain 
/unassign

**Comment by ThreadDao** [comment]
1. it doesn't seem to be fixed on image `master-20220515-338edcd3` `test_delete_insert_same_id_growing` in /testcases/test_delete.py
```
[2022-05-16 10:48:27,710 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 20], kwargs: {} (api_request.py:55)
[2022-05-16 10:48:28,353 - DEBUG - ci_test]: (api_response) : None  (api_request.py:27)
[2022-05-16 10:48:28,396 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [    int64  float varchar                                       float_vector
0       0    0.0       0  [0.14181436381018903, 0.13489670751780697, 0.0...
1       1    1.0       1  [0.14156205921759868, 0.015264006519599253, 0....
2       2    2.0       2  [0.09148366053737855, 0.004494225400454158, 0......, kwargs: {'timeout': 20} (api_request.py:55)
[2022-05-16 10:48:28,781 - DEBUG - ci_test]: (api_response) : (insert count: 100, delete count: 0, upsert count: 0, timestamp: 433237343261425665)  (api_request.py:27)
[2022-05-16 10:48:28,782 - DEBUG - ci_test]: (api_request)  : [Collection.delete] args: ['int64 in [0]', None, 20], kwargs: {} (api_request.py:55)
[2022-05-16 10:48:28,848 - DEBUG - ci_test]: (api_response) : (insert count: 0, delete count: 1, upsert count: 0, timestamp: 433237343288164353)  (api_request.py:27)
[2022-05-16 10:48:28,848 - DEBUG - ci_test]: to_query:False (test_delete.py:762)
[2022-05-16 10:48:28,856 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [   int64  float varchar                                       float_vector
0      0    0.0       0  [0.12836018121933673, 0.045536876385168824, 0...., None], kwargs: {'timeout': 20} (api_request.py:55)
[2022-05-16 10:48:28,945 - DEBUG - ci_test]: (api_response) : (insert count: 1, delete count: 0, upsert count: 0, timestamp: 433237343310970882)  (api_request.py:27)
[2022-05-16 10:48:28,945 - DEBUG - ci_test]: to_flush:True (test_delete.py:769)
[2022-05-16 10:48:32,750 - DEBUG - ci_test]: 101 (test_delete.py:771)
[2022-05-16 10:48:32,754 - DEBUG - ci_test]: (api_request)  : [Collection.query] args: ['int64 in [0]', ['float_vector'], None, 20], kwargs: {} (api_request.py:55)
[2022-05-16 10:48:33,057 - DEBUG - ci_test]: (api_response) : []  (api_request.py:27)
FAILED
```
2. it doesn't seem to be fixed on image `2.0.2-20220513-31a95400`
```
[2022-05-16 11:04:16,076 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None], kwargs: {'timeout': 20} (api_request.py:55)
[2022-05-16 11:04:16,740 - DEBUG - ci_test]: (api_response) : None  (api_request.py:27)
[2022-05-16 11:04:16,778 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [    int64  float                                       float_vector
0       0    0.0  [0.003340548533355106, 0.13838042858803692, 0....
1       1    1.0  [0.054913201447721265, 0.017841021182500798, 0...
2       2    2.0  [0.09753707311815414, 0.049295803071434025, 0....
3       3    3.0  [0.101811......, kwargs: {'timeout': 20} (api_request.py:55)
[2022-05-16 11:04:17,202 - DEBUG - ci_test]: (api_response) : (insert count: 100, delete count: 0, upsert count: 0, timestamp: 433237591880368130)  (api_request.py:27)
[2022-05-16 11:04:17,203 - DEBUG - ci_test]: (api_request)  : [Collection.delete] args: ['int64 in [0]', None, 20], kwargs: {} (api_request.py:55)
[2022-05-16 11:04:17,260 - DEBUG - ci_test]: (api_response) : (insert count: 0, delete count: 1, upsert count: 0, timestamp: 433237591906582531)  (api_request.py:27)
[2022-05-16 11:04:17,261 - DEBUG - ci_test]: to_query:False (test_delete.py:762)
[2022-05-16 11:04:17,275 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [   int64  float                                       float_vector
0      0    0.0  [0.14834974600693318, 0.08488765878743945, 0.0..., None], kwargs: {'timeout': 20} (api_request.py:55)
[2022-05-16 11:04:17,354 - DEBUG - ci_test]: (api_response) : (insert count: 1, delete count: 0, upsert count: 0, timestamp: 433237591932796931)  (api_request.py:27)
[2022-05-16 11:04:17,355 - DEBUG - ci_test]: to_flush:True (test_delete.py:769)
[2022-05-16 11:04:20,631 - DEBUG - ci_test]: 101 (test_delete.py:771)
[2022-05-16 11:04:20,635 - DEBUG - ci_test]: (api_request)  : [Collection.query] args: ['int64 in [0]', ['float_vector'], None, 20], kwargs: {} (api_request.py:55)
[2022-05-16 11:04:20,860 - DEBUG - ci_test]: (api_response) : []  (api_request.py:27)
FAILED
```

**Comment by ThreadDao** [comment]
> Looks like the data is there.
> 
> Test failed because this method crashed:
> 
> https://github.com/milvus-io/milvus/blob/e38c6f6c44b991fb68402094b2158f8c358ebff7/tests/python_client/check/param_check.py#L151

May I ask where is the data returned by the query? Could you post the log? I think the assertion fails because the query returns empty. It's not about this method. 
In the log below, `check_items[exp_res]`represents the expected result, and `query_res` is the actual result, which is not as expected, so the case fails.
```
query_res = [], func_name = 'query'
check_items = {'exp_res': [{'float_vector': [0.1579384606484025, 0.01754486199837335, 0.15751941624097957, 0.1462289854153815, 0.11042587585453352, 0.08468493486250603, ...], 'int64': 0}], 'with_vec': True}
```
